Code/Framework/AzToolsFramework/Tests/FocusMode/EditorFocusModeFixture.cpp

```diff
@@ -0,0 +1,92 @@
+        // shared across the whole engine, if multiple tests are run in parallel, the saving could cause a crash 
+        // in the unit tests.
+        AZ::UserSettingsComponentRequestBus::Broadcast(&AZ::UserSettingsComponentRequests::DisableSaveOnFinalize);
+
+        m_focusModeInterface = AZ::Interface<FocusModeInterface>::Get();
+        ASSERT_TRUE(m_focusModeInterface != nullptr);
+
+        // register a simple component implementing BoundsRequestBus and EditorComponentSelectionRequestsBus
+        GetApplication()->RegisterComponentDescriptor(BoundsTestComponent::CreateDescriptor());
+
+        GenerateTestHierarchy();
+    }
+
+    void EditorFocusModeFixture::GenerateTestHierarchy() 
+    {
+        /*
+        *   City
+        *   |_  Street
+        *       |_  Car
+        *       |   |_ Passenger
+        *       |_  SportsCar
+        *           |_ Passenger
+        */
+
+        m_entityMap[CityEntityName] =       CreateEditorEntity(CityEntityName,          AZ::EntityId());
+        m_entityMap[StreetEntityName] =     CreateEditorEntity(StreetEntityName,        m_entityMap[CityEntityName]);
+        m_entityMap[CarEntityName] =        CreateEditorEntity(CarEntityName,           m_entityMap[StreetEntityName]);
+        m_entityMap[Passenger1EntityName] = CreateEditorEntity(Passenger1EntityName,    m_entityMap[CarEntityName]);
+        m_entityMap[SportsCarEntityName] =  CreateEditorEntity(SportsCarEntityName,     m_entityMap[StreetEntityName]);
+        m_entityMap[Passenger2EntityName] = CreateEditorEntity(Passenger2EntityName,    m_entityMap[SportsCarEntityName]);
+
+        MoveEntityAndAddBoundComponent(m_entityMap[CityEntityName], CityEntityPosition);
+        MoveEntityAndAddBoundComponent(m_entityMap[CarEntityName], CarEntityPosition);
+
+        AzFramework::SetCameraTransform(
+            m_cameraState,
+            AZ::Transform::CreateFromQuaternionAndTranslation(
+                AZ::Quaternion::CreateFromEulerAnglesDegrees(AZ::Vector3(CameraPitch, 0.0f, CameraYaw)), CameraPosition));
+    }
+
+    AZ::EntityId EditorFocusModeFixture::CreateEditorEntity(const char* name, AZ::EntityId parentId)
+    {
+        AZ::Entity* entity = nullptr;
+        UnitTest::CreateDefaultEditorEntity(name, &entity);
+
+        // Parent
+        AZ::TransformBus::Event(entity->GetId(), &AZ::TransformInterface::SetParent, parentId);
+
+        return entity->GetId();
+    }
+
+    void EditorFocusModeFixture::MoveEntityAndAddBoundComponent(AZ::EntityId entityId, AZ::Vector3 position)
+    {
+        AZ::Entity* entity = GetEntityById(entityId);
+        AZ::TransformBus::Event(entityId, &AZ::TransformBus::Events::SetWorldTranslation, position);
+
+        entity->Deactivate();
+        entity->CreateComponent<BoundsTestComponent>();
+        entity->Activate();
+    }
+
+    void EditorFocusModeFixture::TearDownEditorFixtureImpl()
+    {
+    }
+
+    AzToolsFramework::EntityIdList EditorFocusModeFixture::GetSelectedEntities()
```

Comment:
update: This doesn't need to be a member function, it can just be a free function (and I think might already exist in `AzToolsFramework`)

Reply:
Moved selection functions outside the class.